### ISSUE TITLE
Issue 001: fbpick path_tracing core data model と 3D-ready skeleton を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#350]
# Issue 001: fbpick path_tracing core data model と 3D-ready skeleton を追加する

## 背景

coarse GPT は、coarse model の logits を time-axis `log_softmax` score に変換し、その 2D score field 上で global path を選ぶ処理として実装する。MVP は 2D GPT だが、将来 3D GPT に拡張できるように、coarse/infer.py に直接 DP を書かず、`ScoreVolume + Topology + Dispatcher` という共通 API を先に用意する。

Zhang et al. の GPT は 2D array 上で accumulation / backtracking を行い、strain factor で隣接 trace 間の time variation を制約する。3D では inline / crossline の両方向に制約を入れるため、2D の単純な receiver-line 分割だけでは不十分になる。今回の issue では、2D 実装前にこの拡張余地を壊さない module boundary を作る。

## 目的

`fbpick/path_tracing` package を追加し、MVP 2D GPT と将来 3D GPT の共通 data model を定義する。

## 非目的

この issue では以下は行わない。

- DP2D algorithm の本実装
- coarse/infer.py への接続
- `.coarse.npz` diagnostic key の保存
- raw waveform / amplitude ratio / pilot GPT
- `.coarse.npz` だけから疑似 score を作る postprocess CLI

## 変更対象

新規追加:

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/path_tracing/__init__.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/path_tracing/core.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/path_tracing/dp3d.py
packages/seisai-engine/tests/test_fbpick_path_tracing_core.py
```

### 1. `core.py` dataclass

以下を追加する。

```python
from dataclasses import dataclass
from typing import Literal
import numpy as np

PathTracingAlgorithm = Literal[
    "dp2d",
    "dp3d_crossline_smooth_then_inline",
]

ScoreType = Literal["log_softmax"]

TopologyKind = Literal["segment_1d", "grid_2d"]
```

```python
@dataclass(frozen=True)
class GlobalPathTracingCfg:
    enabled: bool = False
    algorithm: PathTracingAlgorithm = "dp2d"
    score_type: ScoreType = "log_softmax"
    max_jump_coarse_samples: int = 4
    min_component_anchors: int = 4
    fallback_short_component_to_argmax: bool = True
    diagnostics: bool = True
```

```python
@dataclass(frozen=True)
class ScoreVolume:
    score: np.ndarray
    prob: np.ndarray | None
    valid_mask: np.ndarray
    axes: tuple[str, ...]
    time_axis: int
```

MVP 2D の想定:

```text
score.shape = (n_anchor_rows, n_time)
axes = ("trace", "time")
time_axis = 1
```

将来 3D の想定:

```text
score.shape = (n_inline, n_crossline, n_time)
axes = ("inline", "crossline", "time")
time_axis = 2
```

```python
@dataclass(frozen=True)
class PathTopology:
    kind: TopologyKind
    component_id: np.ndarray
    valid_mask: np.ndarray
    coordinate_axes: dict[str, np.ndarray] | None = None
```

MVP 2D `segment_1d` では、`component_id.shape == (n_anchor_rows,)` とする。

```python
@dataclass(frozen=True)
class PathTracingResult:
    path_j: np.ndarray
    path_score: np.ndarray
    path_prob: np.ndarray
    component_id: np.ndarray
    valid_mask: np.ndarray
    algorithm: str
```

MVP 2D では以下の形を要求する。

```text
path_j.shape == (n_anchor_rows,)       int64, invalid は -1
path_score.shape == (n_anchor_rows,)   float32, invalid は nan
path_prob.shape == (n_anchor_rows,)    float32, invalid は nan
component_id.shape == (n_anchor_rows,) int64
valid_mask.shape == (n_anchor_rows,)   bool
```

### 2. validation helper

`core.py` に helper を追加する。

```python
def validate_score_volume(volume: ScoreVolume) -> None: ...

def validate_segment_1d_topology(
    topology: PathTopology,
    *,
    n_rows: int,
) -> None: ...
```

`validate_score_volume()` の要件:

- `score` は numpy array。
- `score.ndim >= 2`。
- `score` は floating dtype。
- `score` は finite。
- `time_axis` は有効な axis index。
- `axes` の長さは `score.ndim` と一致。
- `"time"` axis が `time_axis` に対応する。
- `prob is not None` の場合、shape は `score.shape` と一致し、floating dtype, finite, `[0, 1]` 範囲。
- `valid_mask` は bool で、`score` の非 time axes shape と一致する。MVP 2D では `(n_anchor_rows,)`。

`validate_segment_1d_topology()` の要件:

- `topology.kind == "segment_1d"`。
- `component_id` は int dtype, shape `(n_rows,)`。
- `valid_mask` は bool dtype, shape `(n_rows,)`。
- valid row では `component_id >= 0`。
- invalid row は `component_id` が何でもよいが、既存 coarse meta との整合を考え `-1` を推奨する。

### 3. dispatcher skeleton

`core.py` に dispatcher skeleton を追加する。

```python
def trace_global_path(
    score_volume: ScoreVolume,
    topology: PathTopology,
    cfg: GlobalPathTracingCfg,
) -> PathTracingResult:
    ...
```

この issue では:

- `cfg.enabled == False` の場合は `ValueError` を出す。fallback は coarse 側で行う。
- `cfg.algorithm == "dp2d"` はまだ `NotImplementedError`。Issue 002/003 で実装する。
- `cfg.algorithm == "dp3d_crossline_smooth_then_inline"` は `dp3d.trace_dp3d_crossline_smooth_then_inline(...)` に委譲するが、現時点では `NotImplementedError`。
- 未知 algorithm は `ValueError`。

### 4. `dp3d.py` placeholder

```python
def trace_dp3d_crossline_smooth_then_inline(*args, **kwargs):
    raise NotImplementedError(
        "3D GPT is reserved for future work; MVP implements algorithm='dp2d' only"
    )
```

### 5. exports

`path_tracing/__init__.py` で以下を export する。

```python
GlobalPathTracingCfg
PathTopology
PathTracingResult
ScoreVolume
trace_global_path
validate_score_volume
validate_segment_1d_topology
```

## テスト要件

新規:

```text
packages/seisai-engine/tests/test_fbpick_path_tracing_core.py
```

最低限のテスト:

```python
def test_validate_score_volume_accepts_2d_log_score() -> None: ...
def test_validate_score_volume_rejects_nonfinite_score() -> None: ...
def test_validate_score_volume_rejects_prob_shape_mismatch() -> None: ...
def test_validate_segment_1d_topology_accepts_valid_rows() -> None: ...
def test_validate_segment_1d_topology_rejects_negative_valid_component() -> None: ...
def test_trace_global_path_dp3d_placeholder_raises_not_implemented() -> None: ...
```

## 受け入れ条件

- 新 package が import できる。
- `ScoreVolume`, `PathTopology`, `GlobalPathTracingCfg` が stable API として使える。
- 3D algorithm は config / API 上は予約されているが、実行時には明確な `NotImplementedError`。
- 既存 coarse / physics / fine の挙動は変わらない。

## 推奨確認コマンド

```bash
pytest packages/seisai-engine/tests/test_fbpick_path_tracing_core.py -q
python -m compileall packages/seisai-engine/src/seisai_engine/pipelines/fbpick/path_tracing
```
